src/theory/arith/theory_arith.cpp

```diff
@@ -182,12 +183,60 @@ void TheoryArith::postCheck(Effort level)
     // linear solver emitted a conflict or lemma, return
     return;
   }
+  if (d_im.hasSent())
+  {
+    return;
+  }
 
   if (Theory::fullEffort(level))
   {
+    d_arithModelCache.clear();
+    std::set<Node> termSet;
+    collectAssertedTerms(termSet);
+    computeRelevantTerms(termSet);
```

Comment:
You can delete this line.

Reply:
Do you mean ´d_arithModelCache.clear();`? I'd rather keep it: `TheoryArithPrivate::collectModelValues` doesn't do it, and I'd rather not have flying outdated assignments around. (Note that `d_arithModelCache` is not context dependent).

Reply:
No i mean `computeRelevantTerms(termSet);`